LuaSkin/LuaSkin/Skin.m

```diff
@@ -294,6 +338,15 @@ - (void)checkArgs:(int)firstArg, ... {
                 break;
             case LUA_TNUMBER:
                 lsType = LS_TNUMBER;
+
+                if ((spec & lsType) && (spec & LS_TINTEGER)) {
+                    int isInteger ;
+                    lua_tointegerx(_L, idx, &isInteger) ;
+                    if (!isInteger) {
```

Comment:
I think this might break down a little if someone were to specify LS_TNUMBER | LS_TINTEGER. I'm not necessarily advocating that that is a sensible thing to do, it was just a thought that occurred to me. I think given the choice I'd rather document that that is not a sensible thing to do, than make this code more complex.